tests/phpunit/tests/query/searchColumns.php

```diff
@@ -0,0 +1,414 @@
+<?php
+/**
+ * Testing the search columns support in `WP_Query`.
+ *
+ * @package WordPress\UnitTests
+ * @since 6.2.0
+ */
+
+/**
+ * Test cases for the search columns feature.
+ *
+ * @group query
+ * @group search
+ *
+ * @covers WP_Query::parse_search
+ *
+ * @since 6.2.0
+ */
+class Tests_Query_SearchColumns extends WP_UnitTestCase {
+	/**
+	 * The post ID of the first fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid1
+	 */
+	protected static $pid1;
+
+	/**
+	 * The post ID of the second fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid2
+	 */
+	protected static $pid2;
+
+	/**
+	 * The post ID of the third fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid3
+	 */
+	protected static $pid3;
+
+	/**
+	 * Create posts fixtures.
+	 *
+	 * @param WP_UnitTest_Factory $factory The factory instance.
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$pid1 = $factory->post->create(
+			array(
+				'post_status'  => 'publish',
+				'post_title'   => 'foo title',
+				'post_excerpt' => 'foo excerpt',
+				'post_content' => 'foo content',
+			)
+		);
+		self::$pid2 = $factory->post->create(
+			array(
+				'post_status'  => 'publish',
+				'post_title'   => 'bar title',
+				'post_excerpt' => 'foo bar excerpt',
+				'post_content' => 'foo bar content',
+			)
+		);
+
+		self::$pid3 = $factory->post->create(
+			array(
+				'post_status'  => 'publish',
+				'post_title'   => 'baz title',
+				'post_excerpt' => 'baz bar excerpt',
+				'post_content' => 'baz bar foo content',
+			)
+		);
+	}
+
+	/**
+	 * Tests that search uses default search columns when search columns are empty.
+	 *
+	 * @ticket 43867
+	 */
+	public function test_s_should_use_default_search_columns_when_empty_search_columns() {
+		$q = new WP_Query(
+			array(
+				's'              => 'foo',
+				'search_columns' => array(),
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertStringContainsString( 'post_title', $q->request, 'SQL request should contain post_title string.' );
+		$this->assertStringContainsString( 'post_excerpt', $q->request, 'SQL request should contain post_excerpt string.' );
+		$this->assertStringContainsString( 'post_content', $q->request, 'SQL request should contain post_content string.' );
+		$this->assertSameSetsWithIndex( array( self::$pid1, self::$pid2, self::$pid3 ), $q->posts, 'Query results should be equal to the set.' );
```

Comment:
Noting that we keep seeing sporadic test failures (for any given test), and that we _do_ care about the order of elements in this PR, let's see what happens if we change all instances of `assertSameSetsWithIndex()` to `assertSame()`.

Reply:
This seems to fix the test failures. I was investigating locally but couldn't replicate the failures and thought this was just some issue with GitHub actions.